jugaad_data/nse/archives.py

```diff
@@ -133,6 +136,46 @@ def bhavcopy_fo_save(self, dt, dest, skip_if_present=True):
         with open(fname, 'w') as fp:
             fp.write(text)
         return fname
+    
+    @unzip
+    def udiff_bhavcopy_fo_raw(self, dt):
+        """Downloads raw bhavcopy text for a specific date"""
+        dd = dt.strftime('%d')
+        mm = dt.strftime('%m').upper()
+        yyyy = dt.year
+        r = self.get("udiff_bhavcopy_fo", yyyy=yyyy, mm=mm, dd=dd)
+        return r.content
+    
+    def udiff_bhavcopy_fo_save(self, dt, dest, skip_if_present=True):
+        """ Saves Derivatives Bhavcopy to a directory """
+        fmt = "BhavCopy_NSE_FO_0_0_0_%Y%m%d_F_0000.csv"
+        fname = os.path.join(dest, dt.strftime(fmt))
+        if os.path.isfile(fname) and skip_if_present:
+            return fname
+        text = self.udiff_bhavcopy_fo_raw(dt)
+        with open(fname, 'w') as fp:
+            fp.write(text)
+        return fname
+    
+    @unzip
+    def udiff_bhavcopy_raw(self, dt):
+        """Downloads raw bhavcopy text for a specific date"""
+        dd = dt.strftime('%d')
+        mm = dt.strftime('%m').upper()
+        yyyy = dt.year
+        r = self.get("udiff_bhavcopy", yyyy=yyyy, mm=mm, dd=dd)
+        return r.content
+    
+    def udiff_bhavcopy_save(self, dt, dest, skip_if_present=True):
+        """ Saves Derivatives Bhavcopy to a directory """
+        fmt = "BhavCopy_NSE_FO_0_0_0_%Y%m%d_F_0000.csv"
```

Comment:
```suggestion
        fmt = "BhavCopy_NSE_CM_0_0_0_%Y%m%d_F_0000.csv"
```

@kushal-goenka Added a comment

Reply:
thanks! overlooked that file name